src/test/ui/symbol-names/const-generics-demangling.rs

```diff
@@ -1,38 +1,61 @@
 // build-fail
-// compile-flags: -C symbol-mangling-version=v0 --crate-name=c
+// revisions: legacy v0
+// compile-flags: --crate-name=c
+//[legacy]compile-flags: -C symbol-mangling-version=legacy -Z unstable-options
+//    [v0]compile-flags: -C symbol-mangling-version=v0
 // normalize-stderr-test: "c\[.*?\]" -> "c[HASH]"
 #![feature(rustc_attrs)]
 
 pub struct Unsigned<const F: u8>;
 
-#[rustc_symbol_name]
-//~^ ERROR symbol-name(_RMCs
-//~| ERROR demangling(<c[
-//~| ERROR demangling-alt(<c::Unsigned<11>>)
-impl Unsigned<11> {}
+impl Unsigned<11> {
+    #[rustc_symbol_name]
+    //[v0]~^ ERROR symbol-name(_RNvMCs
+    //[v0]~| ERROR demangling(<c[
+    //[v0]~| ERROR demangling-alt(<c::Unsigned<11>>::f)
+    //[legacy]~^^^^ ERROR symbol-name(_ZN1c21Unsigned$LT$11_u8$GT$
+    //[legacy]~|    ERROR demangling(c::Unsigned<11_u8>::f::
+    //[legacy]~|    ERROR demangling-alt(c::Unsigned<11_u8>::f)
+    fn f() {}
+}
 
 pub struct Signed<const F: i16>;
 
-#[rustc_symbol_name]
-//~^ ERROR symbol-name(_RMs_Cs
-//~| ERROR demangling(<c[
-//~| ERROR demangling-alt(<c::Signed<-152>>)
-impl Signed<-152> {}
+impl Signed<-152> {
+    #[rustc_symbol_name]
+    //[v0]~^ ERROR symbol-name(_RNvMs_Cs
+    //[v0]~| ERROR demangling(<c[
+    //[v0]~| ERROR demangling-alt(<c::Signed<-152>>::f)
+    //[legacy]~^^^^ ERROR symbol-name(_ZN1c22Signed$LT$.152_i16$GT$
+    //[legacy]~|    ERROR demangling(c::Signed<.152_i16>::f::
+    //[legacy]~|    ERROR demangling-alt(c::Signed<.152_i16>::f)
```

Comment:
The `.` instead of `-` seems quite surprising, but preexisting. Apparently `-`, `:` and `.` are mangled the same way, and demangled as `.`.